src/Cache.php

```diff
@@ -409,4 +365,111 @@ private function iterableToArray(iterable $iterable): array
     {
         return $iterable instanceof \Traversable ? iterator_to_array($iterable) : (array)$iterable;
     }
+
+
+    /**
+     * Evaluates dependency if it is not null and adds it to the value
+     * @param mixed $value
+     * @param Dependency|null $dependency
+     * @return mixed
+     */
+    private function addDependency($value, ?Dependency $dependency)
+    {
+        if ($dependency === null) {
+            return $value;
+        }
+
+        if (!$dependency->isEvaluated()) {
+            $dependency->evaluateDependency($this);
+        }
+
+        return [$value, $dependency];
+    }
+
+    /**
+     * Checks for the existing values and returns only values that are not in the cache yet.
+     * @param array $values
+     * @return array
+     */
+    private function excludeExistingValues(array $values): array
+    {
+        $existingValues = $this->handler->getMultiple(array_keys($values));
+        foreach ($existingValues as $key => $value) {
+            if ($value !== null) {
+                unset($values[$key]);
+            }
+        }
+
+        return $values;
+    }
+
+    /**
+     * Checks whether value has a dependency and verifies whether it has changed
+     * @param mixed $value
+     * @param mixed $default
+     * @return mixed
+     */
+    private function verifyDependency($value, $default)
```

Comment:
That is not verification. That is getting value considering if dependency was changed i.e. `getValueOrDefaultIfDependencyChanged()` or something like that.